Sparkle/SUHost.m

```diff
@@ -74,7 +75,34 @@ - (BOOL)allowsAutomaticUpdates
     // Note it's very well possible to have the bundle be writable but not be able to write into the parent directory
     // And if the bundle isn't writable, but we can write into the parent directory, we will still need to authorize to replace it
     NSString *bundlePath = [self bundlePath];
-    return [[NSFileManager defaultManager] isWritableFileAtPath:bundlePath.stringByDeletingLastPathComponent] && [[NSFileManager defaultManager] isWritableFileAtPath:bundlePath];
+    if (![[NSFileManager defaultManager] isWritableFileAtPath:bundlePath.stringByDeletingLastPathComponent] || ![[NSFileManager defaultManager] isWritableFileAtPath:bundlePath]) {
+        return NO;
+    }
+    
+    // Just because we have writability access does not mean we can set the correct owner/group silently
+    // Test if we can set the owner/group on a temporarily created file
+    // If we can, then we can probably perform an update without authorization
+    // One place where this matters is if you copy and run an app from /tmp/
+    
+    NSString *tempFilename = @"permission_test" ;
```

Comment:
I suggest using a dot-file to avoid Finder (over)reacting to it. Perhaps `.sparkle_update_permission_test`?